src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java

```diff
@@ -64,35 +93,43 @@ public String s2sSignIn(String s2sName, String s2sSecret, String s2sUrl) {
             .print();
     }
 
-    public void uploadZipFile(
-        CloudBlobContainer container,
+    public Single<BlockBlobUploadResponse> uploadZipFile(
+        ContainerURL container,
         List<String> files,
         String metadataFile,
         final String destZipFilename
     ) throws Exception {
         byte[] zipFile =
             createSignedZipArchiveWithRandomName(files, metadataFile, destZipFilename, TEST_PRIVATE_KEY_DER);
-        CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
-        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
+
+        BlockBlobURL blockBlobURL = container.createBlockBlobURL(destZipFilename);
+        ByteBuffer wrapped = ByteBuffer.wrap(zipFile);
+        Flowable<ByteBuffer> just = Flowable.just(wrapped);
+        return blockBlobURL
+            .upload(just, zipFile.length, null, null, null, Context.NONE);
     }
 
-    public CloudBlockBlob uploadAndLeaseZipFile(
-        CloudBlobContainer container,
+    public String uploadAndLeaseZipFile(
+        ContainerURL container,
         List<String> files,
         String metadataFile,
         String destZipFilename
     ) throws Exception {
-        byte[] zipFile =
-            createSignedZipArchiveWithRandomName(files, metadataFile, destZipFilename, TEST_PRIVATE_KEY_DER);
-        CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
-        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
-        blockBlobReference.acquireLease();
-        return blockBlobReference;
+        BlockBlobURL blockBlobURL = container.createBlockBlobURL(destZipFilename);
+
+        return uploadZipFile(container, files, metadataFile, destZipFilename)
+            .flatMap(r -> blockBlobURL.acquireLease(null, -1, null, Context.NONE))
+            .flatMap(r -> Single.just(r.headers().leaseId()))
+            .blockingGet();
     }
 
-    public boolean storageHasFile(CloudBlobContainer container, String fileName) {
-        return StreamSupport.stream(container.listBlobs().spliterator(), false)
-            .anyMatch(listBlobItem -> listBlobItem.getUri().getPath().contains(fileName));
+    public boolean storageHasFile(ContainerURL container, String fileName) {
+        AzureStorageHelper azureStorageHelper = new AzureStorageHelper(serviceURL);
+
+        return azureStorageHelper.listBlobsLazy(container)
```

Comment:
should have an async and sync version like 
storageHasFile() and storageHasFileSync() or just leave the Sync version out to encourage Rx thinking